kuromoji-core/src/main/java/com/atilika/kuromoji/io/IntegerArrayIO.java

```
@@ -120,4 +104,37 @@ public static void writeSparseArray2D(OutputStream output, int[][] array) throws
         // This negative index serves as an end-of-array marker
         dataOutput.writeInt(-1);
     }
+
+    private static int readIntFromByteChannel(ReadableByteChannel channel) {
+        ByteBuffer intBuffer = ByteBuffer.allocate(INT_BYTES);
+        int result = -1;
+        try {
+            while (intBuffer.hasRemaining() && (channel.read(intBuffer) >= 0)) {
+            }
+            if (!intBuffer.hasRemaining()) {
+                intBuffer.rewind();
+                result = intBuffer.asIntBuffer().get(0);
+            }
+        } catch (Exception ignored) {
+        }
```

Let's just throw this exception to the caller than catch it here.